SingularityService/src/main/java/com/hubspot/singularity/config/SingularityConfiguration.java

```diff
@@ -214,6 +214,11 @@
 
   private long warnIfScheduledJobIsRunningForAtLeastMillis = TimeUnit.DAYS.toMillis(1);
 
+  @JsonProperty("taskExecutionTimeout")
+  @Valid
+  @NotNull
+  private long taskExecutionTimeout = TimeUnit.HOURS.toMillis(12);
```

Comment:
As an addition to that, we should probably have a way to override this at the request level (in `SingularityRequest` object) so that an individual request can enforce a shorter/longer execution time limit if it needs to